src/libs/Navigation/AppNavigator/MainDrawerNavigator.js

```diff
@@ -1,159 +0,0 @@
- * @returns {Object}
- */
-const getInitialReportScreenParams = (reports, ignoreDefaultRooms, policies, openOnAdminRoom) => {
-    const last = ReportUtils.findLastAccessedReport(reports, ignoreDefaultRooms, policies, openOnAdminRoom);
-
-    // Fallback to empty if for some reason reportID cannot be derived - prevents the app from crashing
-    const reportID = lodashGet(last, 'reportID', '');
-    return {reportID: String(reportID)};
-};
-
-class MainDrawerNavigator extends Component {
-    constructor(props) {
-        super(props);
-        this.trackAppStartTiming = this.trackAppStartTiming.bind(this);
-        this.initialParams = getInitialReportScreenParams(
-            props.reports,
-            !Permissions.canUseDefaultRooms(props.betas),
-            props.policies,
-            lodashGet(props, 'route.params.openOnAdminRoom', false),
-        );
-
-        // When we have chat reports the moment this component got created
-        // we know that the data was served from storage/cache
-        this.isFromCache = _.size(props.reports) > 0;
-    }
-
-    shouldComponentUpdate(nextProps) {
-        const initialNextParams = getInitialReportScreenParams(
-            nextProps.reports,
-            !Permissions.canUseDefaultRooms(nextProps.betas),
-            nextProps.policies,
-            lodashGet(nextProps, 'route.params.openOnAdminRoom', false),
-        );
-        if (this.initialParams.reportID === initialNextParams.reportID) {
-            return false;
-        }
-
-        // Update the report screen initial params after the reports are available
-        // to show the correct report instead of the "no access" report.
-        // https://github.com/Expensify/App/issues/12698#issuecomment-1352632883
-        if (!this.initialParams.reportID) {
-            const state = this.props.navigation.getState();
-            const reportScreenKey = lodashGet(state, 'routes[0].state.routes[0].key', '');
-            Navigation.setParams(initialNextParams, reportScreenKey);
-        }
-        this.initialParams = initialNextParams;
-        return true;
-    }
-
-    trackAppStartTiming() {
-        // We only want to report timing events when rendering from cached data
-        if (!this.isFromCache) {
-            return;
-        }
-
-        Timing.end(CONST.TIMING.SIDEBAR_LOADED);
-    }
-
-    render() {
-        return (
-            <BaseDrawerNavigator
-                drawerContent={({navigation, state}) => {
-                    // This state belongs to the drawer so it should always have the ReportScreen as it's initial (and only) route
-                    const reportIDFromRoute = lodashGet(state, ['routes', 0, 'params', 'reportID']);
-                    return (
-                        <SidebarScreen
-                            navigation={navigation}
-                            onLayout={this.trackAppStartTiming}
-                            reportIDFromRoute={reportIDFromRoute}
-                        />
```

Comment:
The removal of `reportIDFromRoute` caused a regression https://github.com/Expensify/App/issues/20703. We added a HOC `withCurrentReportId` that provides `currentReportId` that's meant to be a replacement of `reportIDFromRoute` but we didn't replace all the occurrences of that old prop.